### DISCUSSION TITLE
Is there a way to extend typescript types?

### DISCUSSION THREAD

**Discussion by mc-petry** [TanStack/table#4026] [Q&A] [answered]
I'm extend the table logic with custom methods, for example:
```tsx
function useCustomInstance() {
  const instance = useTableInstance(...)
  return useMemo(() => {
    const helpers = {
      customMethod: () => { ... },
      ...
    }
    return Object.assign(instance, helpers), [instance])
  }
}
```

Than i want to use custom method in column:
```tsx
table.createDataColumn('FirstName', {,
  cell: ({ instance }) => {
    return instance.customMethod()
  }
}),
```

How to extend TableInstance type with my method?

Bad option:
```tsx
return (instance as MyCustomInstance).customMethod()
```
I have a lot of tables with a lots of columns in different places.

**Comment by tannerlinsley** [comment, marked as answer]
Use the tableOptions.meta option. What you pass there will be available on instance.meta. See the editable data example 👍
